great_expectations/experimental/datasources/pandas_s3_datasource.py

```diff
@@ -0,0 +1,290 @@
+        )
+        test_connection_error_message: str = f"""No file in bucket "{self.bucket}" with prefix "{prefix}" matched regular expressions pattern "{regex_pattern}" using deliiter "{delimiter}" for DataAsset "{name}"."""
+        return self.add_asset(
+            asset=asset,
+            data_connector=data_connector,
+            test_connection_error_message=test_connection_error_message,
+        )
+
+    def add_json_asset(
+        self,
+        name: str,
+        regex: Union[str, re.Pattern],
+        prefix: str = "",
+        delimiter: str = "/",
+        max_keys: int = 1000,
+        order_by: Optional[BatchSortersDefinition] = None,
+        **kwargs,  # TODO: update signature to have specific keys & types
+    ) -> JSONAsset:  # type: ignore[valid-type]
+        """Adds a JSON DataAsst to the present "PandasS3Datasource" object.
+
+        Args:
+            name: The name of the csv asset
+            regex: regex pattern that matches csv filenames that is used to label the batches
+            prefix (str): S3 prefix
+            delimiter (str): S3 delimiter
+            max_keys (int): S3 max_keys (default is 1000)
+            order_by: sorting directive via either list[BatchSorter] or "{+|-}key" syntax: +/- (a/de)scending; + default
+            kwargs: Extra keyword arguments should correspond to ``pandas.read_json`` keyword args
+        """
+        regex_pattern: re.Pattern = self.parse_regex_string(regex=regex)
+        order_by_sorters: list[BatchSorter] = self.parse_order_by_sorters(
+            order_by=order_by
+        )
+
+        asset = JSONAsset(
+            name=name,
+            regex=regex_pattern,
+            order_by=order_by_sorters,
+            **kwargs,
+        )
+
+        data_connector: DataConnector = S3DataConnector(
+            datasource_name=self.name,
+            data_asset_name=name,
+            regex=regex_pattern,
+            s3_client=self._s3_client,
+            bucket=self.bucket,
+            prefix=prefix,
+            delimiter=delimiter,
+            max_keys=max_keys,
+            file_path_template_map_fn=S3Url.OBJECT_URL_TEMPLATE.format,
+        )
+        test_connection_error_message: str = f"""No file in bucket "{self.bucket}" with prefix "{prefix}" matched regular expressions pattern "{regex_pattern}" using deliiter "{delimiter}" for DataAsset "{name}"."""
+        return self.add_asset(
+            asset=asset,
+            data_connector=data_connector,
+            test_connection_error_message=test_connection_error_message,
+        )
+
+    def add_parquet_asset(
+        self,
+        name: str,
+        regex: Union[str, re.Pattern],
+        prefix: str = "",
+        delimiter: str = "/",
+        max_keys: int = 1000,
+        order_by: Optional[BatchSortersDefinition] = None,
+        **kwargs,  # TODO: update signature to have specific keys & types
+    ) -> ParquetAsset:  # type: ignore[valid-type]
```

Comment:
Is this just for IDE support, because I'm seeing tab completion in jupyter without this I think.

Reply:
@NathanFarmer Which part of this method definition is raising questions?  Sorry if I am staring at the obvious.  Thanks.